.github/docs/openapi3.txt

```diff
@@ -35,10 +35,10 @@ const (
 	FormatOfStringByte = `(^$|^[a-zA-Z0-9+/\-_]*=*$)`
 
 	// FormatOfStringDate is a RFC3339 date format regexp, for example "2017-07-21".
-	FormatOfStringDate = `^[0-9]{4}-(0[0-9]|10|11|12)-([0-2][0-9]|30|31)$`
+	FormatOfStringDate = `^[0-9]{4}-(0[1-9]|10|11|12)-(0[1-9]|[1-2][0-9]|3[0-1])$`
```

Comment:
I mean, another way to do it would be to rely on go's `time` library (which has parsing logic even more complicated than this under the hood), but then you couldn't publish the exact validation regex in the `.github/doc` files.

Given the regex-based implementation, I think it's a reasonable compromise position go as far as restricting _always_ disallowed inputs and punting issues like "February 30th isn't a thing" onto API implementers.

Reply:
I would also have used a time.Parse, or something equivalent to do not rely on regexp.

But it's already there, so the best we could do is to improve what exists